### PR TITLE
[MC-1498] Add new input in Scan action in InsightVM plugin

### PR DESCRIPTION
**Pull request by dsliwinski-r7**
### Description

Describe the proposed changes:

  - Add new input in Scan action
  - Refactor code in Scan action
  - Add unit tests for Scan action

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "",
    "status": "ok",
    "meta": {},
    "output": {
      "id": 1933,
      "links": [
        {
          "href": "https://example.com/api/3/sites/202/scans?overrideBlackout=false",
          "rel": "self"
        },
        {
          "href": "https://example.com/api/3/scans/1933",
          "rel": "Scan"
        }
      ]
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/rapid7_insightvm:6.1.0 run < tests/scan.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "",
    "status": "ok",
    "meta": {},
    "output": {
      "id": 1934,
      "links": [
        {
          "href": "https://example.com/api/3/sites/213/scans?overrideBlackout=false",
          "rel": "self"
        },
        {
          "href": "https://example.com/api/3/scans/1934",
          "rel": "Scan"
        }
      ]
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/rapid7_insightvm:6.1.0 run < tests/scan_host.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "",
    "status": "ok",
    "meta": {},
    "output": {
      "id": 1935,
      "links": [
        {
          "href": "https://example.com/api/3/sites/212/scans?overrideBlackout=true",
          "rel": "self"
        },
        {
          "href": "https://example.com/api/3/scans/1935",
          "rel": "Scan"
        }
      ]
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/rapid7_insightvm:6.1.0 run < tests/scan_override_blackout.json
</summary>
</details>

### CODE REVIEW

**Comment by dsliwinski-r7** [comment]
PluginValidator and unit tests fails on parts of the plugin that haven't been updated.

**Review by cdankowski-r7 — APPROVED** [review]

**Review by mgrandys-r7 — APPROVED** [review]

File: plugins/rapid7_insightvm/plugin.spec.yaml [hunk 1]
```diff
@@ -1920,17 +1920,26 @@ actions:
         type: "[]string"
         example: ["192.0.2.3", "192.0.2.10-192.0.2.20", "ADSRV.local"]
         required: false
+      override_blackout:
+        title: Override Blackout
+        description: Whether to request for the override of an scan blackout window
```
Comment (dlaverty-r7) [inline plugins/rapid7_insightvm/plugin.spec.yaml:1925]: Suggested reword: 'Set True to override any scan blackout window'

**Review by dlaverty-r7 — APPROVED** [review]
Lgtm just one suggested rewording of description.

File: plugins/rapid7_insightvm/plugin.spec.yaml (hunk 1 above)
Comment (dsliwinski-r7) [reply plugins/rapid7_insightvm/plugin.spec.yaml]: Done

**Review by dlaverty-r7 — APPROVED** [review]

**Pull request merged**